src/node_buffer.cc

```diff
@@ -994,10 +994,12 @@ void IndexOfString(const FunctionCallbackInfo<Value>& args) {
   bool is_forward = args[4]->IsTrue();
 
   const char* haystack = ts_obj_data;
-  const size_t haystack_length = ts_obj_length;
-  // Extended latin-1 characters are 2 bytes in Utf8.
+  // Round down to the nearest multiple of 2 in case of UCS2.
+  const size_t haystack_length = (enc == UCS2) ?
+    ts_obj_length & ~1 : ts_obj_length;
```

Comment:
No worries. Just curious b/c `~1 == -2`. Maybe in that case just ad `  // NOLINT(whitespace/if-one-line)` to the end of the line.